executorCommon/src/main/scala/dx/executor/FileUploader.scala

```diff
@@ -17,4 +30,11 @@ case class SerialFileUploader(dxApi: DxApi = DxApi.get) extends FileUploader {
   def upload(files: Set[Path]): Map[Path, DxFile] = {
     files.map(path => path -> dxApi.uploadFile(path)).toMap
   }
+
+  def upload(files: Map[Path, String]): Map[Path, DxFile] = {
```

Comment:
Would it make sense to implement (here or in another commit) `upload` with a `wait_on_close` option, similar to dxpy. This would eliminate the need for a 10s wait in the test, for example (it's related to https://jira.internal.dnanexus.com/browse/PTFM-24239 as well)

Reply:
that's a good suggestion - however, the ultimate goal is to use upload agent to manage uploads. I have the code ready in a branch, I am just waiting for DEVEX-1814 to be fixed.